### PR TITLE
Feature/sycl rebase

### PR DESCRIPTION
**Pull request by reguly**
@aowenson the changes are fairly minor, mainly around the makefile. How should we handle the instrumentation in the long run?

### CODE REVIEW

**Comment by aowenson** [comment]
@reguly There are a few instances of dead code (e.g. commented-out), can you delete those.

What do you mean by "instrumentation"? PAPI?

File: Makefile [hunk 1]
```diff
@@ -360,6 +446,33 @@ $(BIN_DIR)/mgcfd_seq: $(OP2_SEQ_OBJECTS)
...
+	$(SYCLCXX) $(CXXFLAGS) -DSYCL $(SYCL_FLAGS) $(SYCL_INCLUDES) $(OPTIMISE) $(MGCFD_INCS) \
+	    $(OP2_INC) $(HDF5_INC) $(PARMETIS_INC) $(PTSCOTCH_INC) -I$(MPI_INSTALL_PATH)/include/ \
+		-c -o $@ $(SRC_DIR)/../sycl/_kernels.cpp
+$(BIN_DIR)/mgcfd_sycl: $(OP2_SYCL_OBJECTS)
+	mkdir -p $(BIN_DIR)
+	$(SYCLCXX) $(CXXFLAGS) $(SYCL_FLAGS) $(SYCL_LIBS) $(OPTIMISE) $^ $(MGCFD_LIBS) \
+		-lm $(OP2_LIB) -lop2_sycl $(SYCL_LINK_SEQ) -lop2_hdf5 $(HDF5_LIB) \
+		-o $@
+
+#$(BIN_DIR)/mgcfd_sycl: src_op/euler3d_cpu_double_op.cpp sycl/_kernels.cpp
```
Comment (aowenson) [inline Makefile:466]: dead code?

File: src/euler3d_cpu_double.cpp [hunk 2]
```diff
@@ -10,7 +10,7 @@
 #include <fstream>
 #include <cmath>
 #include <string>
-#include <omp.h>
+//#include <omp.h>
```
Comment (aowenson) [inline src/euler3d_cpu_double.cpp:13]: just delete

File: src/inlined_funcs.h [hunk 3]
```diff
@@ -11,6 +11,8 @@
 
 #define DEBUGGABLE_ABORT fprintf(stderr, "%s:%d\n", __FILE__, __LINE__); fflush(stderr); fflush(stdout); exit(EXIT_FAILURE);
 
+//#define cl::sycl::cbrt(a) cl::sycl::pow(a,1.0/3.0)
```
Comment (aowenson) [inline src/inlined_funcs.h:14]: dead code?

File: src/structures.h [hunk 4]
```diff
@@ -1,13 +1,15 @@
 #ifndef STRUCTURES_H
 #define STRUCTURES_H
 
-#ifndef __CUDACC__
+#if !defined(__HIPSYCL__) && !defined(__CUDACC__)
+//#ifndef __CUDACC__
```
Comment (aowenson) [inline src/structures.h:5]: delete

**Review by aowenson — COMMENTED** [review]
tagged dead code

**Comment by reguly** [comment]
Ah, sorry about these!

regarding instrumentation - yes, the PAPI calls, which are not auto-generated by the translator. Any code generation step will overwrite those.

**Comment by aowenson** [comment]
Ah yes, that really should be automated. Currently I just manually restore after a regen.
I've done something similar with HYDRA instrumentation recently, I should be able to port the script.

**Comment by aowenson** [comment]
Does instrumentation need to be resolved for the merge? Currently I don't see any changes to non-sycl backends, nor src_op/euler, so I would say no.

**Comment by reguly** [comment]
No, we do not need to resolve this right now

**Pull request merged**